linera-execution/src/execution.rs

```diff
@@ -569,16 +564,12 @@ where
             })
             .expect("Service runtime thread should only stop when `request_sender` is dropped");
 
+        // manual fusing.
```

Comment:
I think we could simplify this and remove the `loop`, if we `use std::pin::pin;` and `use futures::future::{self, Either};`:

```rust
let request_handler = pin!(self.handle_requests(incoming_execution_requests));

let outcome = match future::select(outcome_receiver, request_handler).await {
    Either::Left((outcome, _)) => outcome,
    Either::Right((result, outcome_receiver)) => {
        result?;
        outcome_receiver.await
    }
};

outcome.map_err(|_| ExecutionError::MissingRuntimeResponse)?
```